src/phpdoc/type_parser_test.go

```diff
@@ -109,6 +109,11 @@ func TestParser(t *testing.T) {
 		{`shape{s?: string}`, `GenericBrace="shape{s?: string}"{Name="shape" KeyVal="s?: string"{Optional="s?"{Name="s"} Name="string"}}`},
 		{`foo(A):B`, `KeyVal="foo(A):B"{GenericParen="foo(A)"{Name="foo" Name="A"} Name="B"}`},
 
+		// MemberType.
+		{`\Foo::CONST`, `MemberType="\Foo::CONST"{Class="\Foo" Member="CONST"}`},
+		{`\Foo::CONST|\Foo::CONST2`, `Union="\Foo::CONST|\Foo::CONST2"{MemberType="\Foo::CONST"{Class="\Foo" Member="CONST"} MemberType="\Foo::CONST2"{Class="\Foo" Member="CONST2"}}`},
+		{`\Foo::$a|\Foo::CONST2`, `Union="\Foo::$a|\Foo::CONST2"{MemberType="\Foo::$a"{Class="\Foo" Member="$a"} MemberType="\Foo::CONST2"{Class="\Foo" Member="CONST2"}}`},
```

Comment:
You changed the precedence values. Could you add more tests that cover all combinations of the new `::` operator with all existing ones with updated precedence?